### DISCUSSION TITLE
Unable to install Next.js dependencies

### DISCUSSION THREAD

**Discussion by justintemps** [vercel/next.js#19736] [Help] [answered]
I'm trying to debug an issue that I'm having with Next.js, so I've cloned the canary branch and am following the instructions in [contributing](https://github.com/vercel/next.js/blob/canary/contributing.md) but am unable to install the dependencies.

At a certain point after running `yarn`, Lerna executes

```
$ tsc --declaration --emitDeclarationOnly --declarationDir dist
```

which triggers a cascade of TypeScript errors:

```
./../node_modules/@types/react/index.d.ts(2981,14): error TS2300: Duplicate identifier 'LibraryManagedAttributes'.
../../node_modules/@types/react/index.d.ts(2992,13): error TS2717: Subsequent property declarations must have the same type.  Property 'a' must be of type 'DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>', but here has type 'DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>'.
...
```

followed by:

```
lerna ERR! yarn run prepublish stderr:
error Command failed with exit code 2.
error Command failed with exit code 2.

lerna ERR! yarn run prepublish exited 2 in 'next'
error An unexpected error occurred: "Command failed.
Exit code: 2
Command: sh
Arguments: -c lerna run prepublish
Directory: /mydirectory
Output:
".
```

**Comment by jamesmosier** [comment, marked as answer]
The dependencies should be installed at this point. It looks like `"types"` is run [here](https://github.com/vercel/next.js/blob/26634c475e7a5c291f6a5abee3fc47afd95a4776/packages/next/package.json#L51-L51) and that is run via `prepublish` above it which is triggered via the `yarn install` command for each `package` in the `packages` folder.

Are you able to run `yarn dev` or even run some of the tests (I'd recommend only running 1 suite at a time).

**Reply by justintemps** [reply]
Oh hey it looks like dependencies are installed then. So is it common for lerna to error out at the point or am I just crazy for trying to install all the dependencies at once. First time in monorepo, sorry.

**Reply by jamesmosier** [reply]
I wouldn't expect it to error out. I was able to run `yarn install` in the root folder and it would install all the deps and then run the typechecking. Sorry I'm not much more help 😞

**Reply by justintemps** [reply]
No that actually helps a lot @jamesmosier, thanks very much. If I can bother you with a quick follow up, trying to understand some behaviour in Next.js and can't figure out the right approach for debugging it. If I follow the [contribution guidelines](https://github.com/vercel/next.js/blob/canary/contributing.md), I can run my app with a locally compiled version of Next.js, but that's not much good for debugging. I also tried to use `npm link` but can't figure out how to manage symlinked dependencies (Next can't find react, or react-dom, etc.). Any advice or resources you could offer would be hugely appreciated.

**Reply by jamesmosier** [reply]
I haven't run this personally, but have you tried out the [debug](https://github.com/vercel/next.js/blob/20949612dfb56e44a623498eb3237a69177229e5/package.json#L36) npm script? 

In theory (at least from looking at it) that should allow you to attach breakpoints when running Next locally in your project.

There is also [this answer](https://github.com/vercel/next.js/discussions/13272#discussioncomment-18012) that might help out a bit.

I usually just run `yarn dev` in the Next repo and then setup my local project according to the [running Next in a local project](https://github.com/vercel/next.js/blob/canary/contributing.md#running-your-own-app-with-locally-compiled-version-of-nextjs) instructions and then `console.log` for my debugging 😄  And then `yarn install --force` in my local project whenever I need to update it after making a change in Next. Not the best approach for debugging so if the npm script works out for you i'd be interested to hear!

**Reply by justintemps** [reply]
Thanks so much! Actually the approach of running `yarn dev` in Next and installing into my local project works great and I was even able to solve the type errors problem. 

However, I have one more question for you. The problem I'm having only happens in production. So I wanted to make a chance in Next and then do `next build && next start` in my local project. My only problem is that when I do that for some reason Webpack(???) can't find next packages. For instance, I get:

```
Type error: Module '"../../node_modules/next/router"' has no exported member 'useRouter'. Did you mean to use 'import useRouter from "../../node_modules/next/router"' instead?
```

This seems odd because I checked the node_modules and `/node_modules/next/router` does export useRouter. Off the top of your head, do you have any idea why that could be happening?

If it isn't obvious enough, I've never tried debugging an issue with a package like this before and suppose I dove in the deep end with next.

**Reply by jamesmosier** [reply]
I'm not sure unfortunately 😞
